src/OpenLoco/Vehicles/VehicleBody.cpp

```diff
@@ -255,30 +255,25 @@ namespace OpenLoco::Vehicles
     // 0x004AC255
     void VehicleBody::sub_4AC255(VehicleBogie* back_bogie, VehicleBogie* front_bogie)
     {
-        Map::map_pos3 loc = {
-            static_cast<int16_t>((front_bogie->x + back_bogie->x) / 2),
-            static_cast<int16_t>((front_bogie->y + back_bogie->y) / 2),
-            static_cast<int16_t>((front_bogie->z + back_bogie->z) / 2)
-        };
-        moveTo(loc);
+        Map::map_pos3 midPoint = (front_bogie->position - back_bogie->position) / 2;
+        moveTo(midPoint);
 
         if (object_sprite_type == 0xFF)
             return;
 
-        auto distanceBetweenBogies = Math::Vector::distance(Map::map_pos{ front_bogie->x, front_bogie->y }, Map::map_pos{ back_bogie->x, back_bogie->y });
-
+        auto distanceBetweenBogies = Math::Vector::distance(front_bogie->position, back_bogie->position);
         auto vehObj = object();
         if (vehObj->bodySprites[object_sprite_type].flags & BodySpriteFlags::hasSteepSprites)
         {
-            sprite_pitch = updateSpritePitchSteepSlopes(distanceBetweenBogies, front_bogie->z - back_bogie->z);
+            sprite_pitch = updateSpritePitchSteepSlopes(distanceBetweenBogies, front_bogie->position.z - back_bogie->position.z);
         }
         else
         {
-            sprite_pitch = updateSpritePitch(distanceBetweenBogies, front_bogie->z - back_bogie->z);
+            sprite_pitch = updateSpritePitch(distanceBetweenBogies, front_bogie->position.z - back_bogie->position.z);
         }
 
-        auto xDiff = front_bogie->x - back_bogie->x;
-        auto yDiff = front_bogie->y - back_bogie->y;
+        auto xDiff = front_bogie->position.x - back_bogie->position.x;
+        auto yDiff = front_bogie->position.y - back_bogie->position.y;
```

Comment:
This can be simplified as
```suggestion
        auto diff = front_bogie->position - back_bogie->position;
```
This will however shift it to calculateYaw0FromVector